### PR TITLE
Backport pull request #1623 and bump Python version

### PR DESCRIPTION
**Pull request by jougs**

### CODE REVIEW

**Comment by jougs** [comment]
@jasperalbers: I guess this is ready for review and merge.

File: .travis.yml [hunk 1]
```diff
@@ -40,25 +40,25 @@ stages:
...
     - stage: MPI-Threading-Python
-      python: 3.6.7
+      python: 3.8.2
       env: xTHREADING=0 xMPI=0 xGSL=0 xLIBNEUROSIM=0 xLTDL=0 xREADLINE=0 xLIBBOOST=1 xPYTHON=1 xMUSIC=0 xSTATIC_ANALYSIS=0 xRUN_BUILD_AND_TESTSUITE=1 CACHE_NAME=JOB   # only Python
     - stage: Python-Full-build
-      python: 3.6.7
+      python: 3.8.2
       env: xTHREADING=1 xMPI=1 xGSL=1 xLIBNEUROSIM=1 xLTDL=1 xREADLINE=1 xLIBBOOST=1 xPYTHON=1 xMUSIC=1 xSTATIC_ANALYSIS=0 xRUN_BUILD_AND_TESTSUITE=1 CACHE_NAME=JOB   # full
     - stage: Clang7
       language: cpp
```
Comment (jasperalbers) [inline .travis.yml:64]: A few lines further down (line 68, but I can't leave a comment that far down) the code specifies xcode10.3. The original PR #1623 specifies xcode12, while the current xcode version on my MacBook running MacOS 10.15.6 shows xcode11.6 to be the newest version. Not sure whether this discrepancy is intentional or not.

**Review by jasperalbers — COMMENTED** [review]
I left a minor comment about the xcode version, the rest looks fine to me.

File: .travis.yml (hunk 1 above)
Comment (jougs) [reply .travis.yml:64]: I did not want to go too far with my backport, so I left this change out. I think the code is good as is and can be merged.

**Review by jasperalbers — APPROVED** [review]

**Pull request merged**